src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -287,7 +287,7 @@ public void Remove(object key)
 
         private void RemoveEntry(CacheEntry entry)
         {
-            if (EntriesCollection.Remove(new KeyValuePair<object, CacheEntry>(entry.Key, entry)))
+            if (_entries.TryRemove(entry.Key, out _))
```

Comment:
This is not necessarily the same thing.  The previous version would only remove the entry from the ConcurrentDictionary if the supplied entry's value matched what is currently specified for the key.  The new version ignores the value and will remove the key from the dictionary regardless of its value.

Reply:
The PR is failing CI exactly because of this issue, e.g. the:
https://github.com/dotnet/runtime/blob/0c9094eae19e8b539d571b7c9e8646805b12ef51/src/libraries/Microsoft.Extensions.Caching.Memory/tests/MemoryCacheSetAndRemoveTests.cs#L566
test is failing.  I'm going to close the PR as a result.  But thanks for the interest.

Reply:
@stephentoub Thank you